app/assets/javascripts/notebook/magic/tsChart.coffee

```diff
@@ -11,8 +11,14 @@ define([
     chart_container = $("<div>").addClass("custom-c3-chart").attr("width", w+"px").attr("height", h+"px")
     chart_container.attr("id", "custom-c3-chart-"+@genId).appendTo(container)
 
-    data = {
-    }
+    data = {}
+
+    # Datetime data are coming as formatted strings. We need to tell about the format to c3 lib. But we still want
+    # to support opening old notebooks with old format (timestamp in miliseconds) without need to regenerate the data.
+    sampleTimeData = @dataInit[0][options.x]
```

Comment:
wouldn't this fail if `@dataInit` is an empty list (i.e. there were no result rows) ?